### PR TITLE
Fix #1898 and potential bug

### LINKED ISSUE #1898: Delivery reports needs fixing, again
There is something wrong with the fix of the fix apparently, and the link to delivery reports doesn't work.

### PR DESCRIPTION
**Pull request by northwestwitch**
Bug fix #1898

**How to test**:
1. Add analyses and delivery reports from some real case to a test case on stage
1. Check that the link to the delivery reports work

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

File: scout/server/blueprints/cases/templates/cases/collapsible_actionbar.html [hunk 1]
```diff
@@ -411,14 +411,14 @@ <h5 class="modal-title" id="archiveCaseLabel">Confirm archive case</h5>
 </div>
 {% endmacro %}
 
-{% macro delivery_report(institute, case, date) %}
+{% macro delivery_report(institute, case, date=None) %}
```
Comment (dnil) [inline scout/server/blueprints/cases/templates/cases/collapsible_actionbar.html:414]: :+1:

File: scout/server/blueprints/cases/views.py [hunk 2]
```diff
@@ -1124,7 +1124,7 @@ def delivery_report(institute_id, case_name):
     if date_str:
         delivery_report = None
         analysis_date = parse_date(date_str)
-        for analysis_data in case_obj["analyses"]:
+        for analysis_data in case_obj.get("analyses", []):
```
Comment (dnil) [inline scout/server/blueprints/cases/views.py:1127]: :+1:

File: scout/server/blueprints/cases/templates/cases/collapsible_actionbar.html [hunk 3]
```diff
@@ -81,12 +81,12 @@
...
     </div>
     {% if case.delivery_report %}
-      {{ delivery_report(institute._id, case.display_name, case.analysis_date.date()) }}
+      {{ delivery_report(institute._id, case) }}
     {% endif %}
     {% if case.analyses %}
       {% for analysis in case.analyses %}
         {% if analysis.delivery_report and analysis.delivery_report != case.delivery_report %}
-          {{ delivery_report( institute._id, case.display_name, analysis.date.date() ) }}
+          {{ delivery_report( institute._id, case, analysis.date.date() ) }}
```
Comment (dnil) [inline scout/server/blueprints/cases/templates/cases/collapsible_actionbar.html:89]: 👀 Yep!

**Comment by dnil** [comment]
Ok, now the changelog will be easier to update!

**Comment by northwestwitch** [comment]
Wait it's not working yet, I'm testing on stage and those error emails are the problem

**Comment by dnil** [comment]
Yes, I saw - str.date() sth. I'll copy over a couple of old reports to stage in the meantime.

**Comment by northwestwitch** [comment]
> Yes, I saw - str.date() sth. I'll copy over a couple of old reports to stage in the meantime.

I did it, look at this case https://scout-stage.scilifelab.se/cust000/643594-300M

**Comment by dnil** [comment]
This one is already working well now: 
https://scout-stage.scilifelab.se/cust002/F0015031
but it is a single report, modern style db entry with both analyses and case["delivery_report"].

**Comment by dnil** [comment]
I also made one with a working report, linked ok and in modern style, but with an additional null analysis:
https://scout-stage.scilifelab.se/cust002/F0027117

**Comment by dnil** [comment]
Just to be on the safe side I've also modified this one to recreate an analog of the original issue:
https://scout-stage.scilifelab.se/cust002/F0005295-2

case["delivery_report"] exists, but analyses entry for the present case["analysis_date"] has null delivery_report.

**Comment by northwestwitch** [comment]
> https://scout-stage.scilifelab.se/cust002/F0005295-2

Works. So far only the cases with old reports that are non null don't work

**Comment by dnil** [comment]
That actually worked for the old case; only the filename for that particular case was one char off. My bad.😔 Unless you checked the previous error messages meticulously I would guess at least a couple of them also did work.

**Comment by dnil** [comment]
All tested scenarios now work. I will go ahead and merge this now to have a working version, but  lets do the release and deployment with fresh heads.

**Pull request merged**